shim.go

```diff
@@ -22,57 +22,40 @@ import (
 	vc "github.com/containers/virtcontainers"
 )
 
-// ShimInfo gathers information needed by the shim.
-type ShimInfo struct {
-	Path  string
-	Token string
-	IP    string
-	Port  string
-}
-
-func startShim(pod *vc.Pod) (int, error) {
-	containers := pod.GetContainers()
-
-	if len(containers) != 1 {
-		return -1, fmt.Errorf("Container list from pod is wrong, expecting only one container")
+func startShim(process *vc.Process) (int, error) {
+	if process.Token == "" {
+		return -1, fmt.Errorf("Token cannot be empty")
 	}
-	container := containers[0]
 
-	shimInfo, err := getShimInfo(container)
-	if err != nil {
-		return -1, err
+	url := process.URL
+	if url == "" {
+		url = defaultProxyURL
```

Comment:
It might be useful to add a debug log call to show which url is actually being used.